### PR TITLE
Devdocs: serve the content of README.md files from the search index module

### PR DESCRIPTION
**Pull request by jsnajdr**
While working on #38190 and updating the _devdocs/bin/generate-devdocs-index_ script, I discovered that in addition to the search index, we also store the text content of the indexed files in the generated `search-index.js` module. It's used to generate snippets (that contain the searched term) for the search results. This text content is plain text with all markup stripped.

If we stored the raw Markdown content to this module, too, we could use it to serve the _README.md_ files content in the `/devdocs/service/content` endpoint. This PR implements exactly that. Currently it's served from the Calypso source tree that needs to be present at runtime.

I think this is a better and more elegant solution than moving the Markdown content to the client webpack bundle that @ockham proposes in #35833. Mainly because it doesn't inflate the webpack manifest with 800+ new chunks.

It also makes the Calypso server more self-contained: soon we'll need only the `build/` and `public/` directory to run the Calypso server, and not the whole source tree.

#### Changes proposed in this Pull Request

1. Move the generating script to the `/bin` directory and give it a `.js` suffix. That's a better location and the `.js` extension will make it Prettier-ed and ESLint-ed.
2. In `package.json`, make sure that the `build-devdocs:*` tasks are finished before `build-server` starts. `build-server` uses the files generated by `build-devdocs:*`. Until now, they were running in parallel and there was a race condition present.
3. In `package.json`, simplify and rename the `build-devdocs:index` task. The script doesn't need any special environment variables and can be run with a simple `node` command.
4. In the `generate-devdocs-index.js` script, format it with Prettier and fix ESLint errors. And also store the raw content as a `content` property of the generated document records.
5. Finally, in the implemenation of the `/devdocs/service/endpoint`, use `searchIndex.documents[ idx ].content` to serve the raw content. And remove the FS access.

#### Testing instructions

- test that the README search at `/devdocs` works correctly
- test that the `<ReadmeViewer>` component that's used across devdocs shows the README documents as expected.

### CODE REVIEW

File: server/devdocs/index.js [hunk 1]
```diff
@@ -211,8 +213,8 @@ module.exports = function() {
 	// return the content of a document in the given format (assumes that the document is in
 	// markdown format)
 	app.get( '/devdocs/service/content', ( request, response ) => {
-		let path = request.query.path;
-		const format = request.query.format || 'html';
+		let { path } = request.query;
+		const { format = 'html' } = request.query.format;
```
Comment (ockham) [inline server/devdocs/index.js]: Shouldn't this be

```suggestion
		const { format = 'html' } = request.query;
```

?

**Review by ockham — CHANGES_REQUESTED** [review]
Nice! This has a much smaller footprint than mine!

Unfortunately, none of the docs in `devdocs/docs/` are currently working (The Calypso Guide, Accessibility, Color, Icons); same for Contributing (`.github/CONTRIBUTING.md` -- maybe see #36167?)

(Typography, which is in `devdocs/`, does work, as doe component and blocks docs -- except for the ones that we've moved into `packages/components`, but that's a known and pre-existing issue.)

File: server/devdocs/index.js (hunk 1 above)
Comment (ockham) [reply server/devdocs/index.js]: My console seems to agree :grimacing: 

```
TypeError: Cannot destructure property `format` of 'undefined' or 'null'.
```

Comment (jsnajdr) [reply server/devdocs/index.js]: Yes, I ruined the code when trying to migrate it to fancy ES6 destructuring! Fixed up in https://github.com/Automattic/wp-calypso/pull/38784/commits/04bdaec8b1032eb15c38e0e1ce2a208f92e933bf

**Comment by jsnajdr** [comment]
> Unfortunately, none of the docs in `devdocs/docs/` are currently working

Oh, I misunderstood what `path.relative()` really does and didn't test the patch sufficiently 🙁 Fixed in https://github.com/Automattic/wp-calypso/pull/38784/commits/f5aac84d22be277af7ddae22fbdd825950491585

> except for the ones that we've moved into `packages/components`

Files in `packages/` were not part of the search index until now -- that's fixed by this added line: https://github.com/Automattic/wp-calypso/pull/38784/files#diff-8434269b829266816be9c69918de4979R24

Displaying their content now works, too -- I don't see any regressions or bugs for `packages/components/`

All issues from the review should be addressed now.

**Review by ockham — APPROVED** [review]
Thanks @jsnajdr, this is working great now!

**Pull request merged**